src/Uno.UWPSyncGenerator/Generator.cs

```diff
@@ -1746,40 +1755,36 @@ private static Compilation LoadProject(string projectFile, string targetFramewor
 				return compilation;
 			}
 
-			return _projects[key] = InnerLoadProject(projectFile, targetFramework);
+			return _projects[key] = await InnerLoadProject(projectFile, targetFramework);
 		}
 
-		private static Compilation LoadUWPReferenceProject(string referencesFile)
+		private static async Task<Compilation> LoadUWPReferenceProject(string referencesFile)
 		{
 			var ws = new AdhocWorkspace();
 
 			var p = ws.AddProject("uwpref", LanguageNames.CSharp);
-
-			foreach (var reference in File.ReadAllLines(referencesFile))
-			{
-				p = p.AddMetadataReference(MetadataReference.CreateFromFile(reference));
-			}
-
-			return p.GetCompilationAsync().Result;
+			p = p.AddMetadataReferences(File.ReadAllLines(referencesFile).Select(reference => MetadataReference.CreateFromFile(reference)));
+			return await p.GetCompilationAsync();
 		}
 
-		private static Compilation InnerLoadProject(string projectFile, string targetFramework = null)
+		private static async Task<Compilation> InnerLoadProject(string projectFile, string targetFramework = null)
 		{
 			Console.WriteLine($"Loading for {targetFramework}: {Path.GetFileName(projectFile)}");
 
 			var properties = new Dictionary<string, string>
-							{
-								// { "VisualStudioVersion", "15.0" },
-								// { "Configuration", "Debug" },
-								//{ "BuildingInsideVisualStudio", "true" },
-								{ "SkipUnoResourceGeneration", "true" }, // Required to avoid loading a non-existent task
-								{ "DocsGeneration", "true" }, // Detect that source generation is running
-								{ "LangVersion", CSharpLangVersion },
-								{ "NoBuild", "True" },
-								//{ "DesignTimeBuild", "true" },
-								//{ "UseHostCompilerIfAvailable", "false" },
-								//{ "UseSharedCompilation", "false" },
-							};
+			{
+				// { "VisualStudioVersion", "15.0" },
+				// { "Configuration", "Debug" },
+				//{ "BuildingInsideVisualStudio", "true" },
+				{ "SkipUnoResourceGeneration", "true" }, // Required to avoid loading a non-existent task
+				{ "DocsGeneration", "true" }, // Detect that source generation is running
+				{ "LangVersion", CSharpLangVersion },
+				{ "NoBuild", "True" },
+				{ "RunAnalyzers", "false" }
```

Comment:
I have no idea if they are run or not, but just to make sure :)

Reply:
I'd think that with the `NoBuild` parameter, they should not :) But yep, does not hurt.